sandboxnu/graduatenu
# Setup DB migrations for both development and prod

# Description

- Migrations allow us to deal with changes to our database systematically and sensitively. This is important once we are a production application.
- Previously, we used a typeorm feature that allowed us to automatically synchronize the database columns/tables based on the current entity definition. This is nice but having "automatic" stuff happen in production environments is hella sketch.
- From now on, any change to the database will go through a process:
  - Make entity change
  - Generate migration file that details how to take DB from current state -> new state and vice versa.
  - Run migration file locally to update local db.
  - Open PR, have someone review the migration.
  - Merge PR -> once deployed the new migration will be run on the production db before starting the new server.

Closes #521 

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Development flow test:
See README of API for yarn commands to run 
- Generated the 1st migration that sets up the DB according to our current entities.
- Ran the migration against a local db with no tables.
- App works fine with new tables.
- Make a test change to the db. Repeat process and make sure everything works. Remove test change.

Production test:
- Build backend docker image
- Spin up container running the image, ensure it *attempts* to run the migrations. This will fail cause we don't have it setup for the db and backend to run in docker at the same time locally yet. This is the best I can do for now, fingers crossed I don't take down staging.

# Checklist:
